web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,457 @@
+  }
+];
+
+// Names of the columns for the policy actions.
+const columnNames = {
+  device: N_("Used device"),
+  content: N_("Current content"),
+  size: N_("Size"),
+  action: N_("Action")
+};
+
+/**
+ * Indicates whether a device is a drive (disk, RAID).
+ * @function
+ *
+ * @param {StorageDevice} device
+ * @returns {boolean}
+ */
+const isDrive = (device) => Object.keys(device).includes("vendor");
+
+/**
+ * Column content with the description of a device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceDescriptionColumn = ({ device }) => {
+  return (
+    <>
+      <div>{device.name}</div>
+      <If
+        condition={isDrive(device)}
+        then={<div className="fs-small">{`${device.vendor} ${device.model}`}</div>}
+      />
+    </>
+  );
+};
+
+/**
+ * Column content with information about the current content of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceContentColumn = ({ device }) => {
+  const PartitionTableContent = () => {
+    const numPartitions = device.partitionTable.partitions.length;
+
+    return (
+      <>
+        <div>
+          {sprintf(n_("%d partition", "%d partitions", numPartitions), numPartitions)}
+        </div>
+        <div className="fs-small">
+          {sprintf(_("%s partition table"), device.partitionTable.type.toUpperCase())}
+        </div>
+      </>
+    );
+  };
+
+  const BlockContent = () => {
+    const systems = device.systems;
+    const filesystem = device.filesystem;
+
+    const content = () => {
+      if (systems.length > 0) return systems.join(", ");
+      if (device.filesystem?.isEFI) return _("EFI");
+
+      return _("Not identified");
+    };
+
+    return (
+      <>
+        <div>{content()}</div>
+        <If
+          condition={filesystem}
+          then={
+            <div className="fs-small">
+              {sprintf(_("%s file system"), filesystem?.type)}
+            </div>
+          }
+        />
+      </>
+    );
+  };
+
+  return (device.partitionTable ? <PartitionTableContent /> : <BlockContent />);
+};
+
+/**
+ * Column content with information about the size of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceSizeColumn = ({ device }) => {
+  const UnusedSize = () => {
+    const used = device.partitionTable?.partitions.reduce((s, p) => s + p.size, 0) || 0;
+    const unused = device.size - used;
+
+    if (unused === 0) return null;
+
+    return (
+      <div className="fs-small">
+        {sprintf(_("%s unused"), deviceSize(unused))}
+      </div>
+    );
+  };
+
+  const RecoverableSize = () => {
+    const size = device.recoverableSize;
+    let text;
+
+    if (size === 0)
+      text = _("No recoverable space");
+    else
+      text = sprintf(_("%s recoverable"), deviceSize(device.recoverableSize));
+
+    return <div className="fs-small">{text}</div>;
```

Comment:
Note for another iteration: try something more than simply smaller text. Also check how these "less important" information should be _structured_ in terms of a11y.